modules/pinger/src/main/java/org/hawkular/component/pinger/MetricPublisher.java

```diff
@@ -47,72 +50,110 @@
 @Stateless
 public class MetricPublisher {
 
-    @javax.annotation.Resource( lookup = "java:/topic/HawkularMetricData")
+    private static void addDataItem(List<Map<String, Object>> mMetrics, String resourceId, long timestamp,
+            Number value, String name) {
+        Map<String, Number> dataMap = new HashMap<>(4);
+        dataMap.put("timestamp", timestamp);
+        dataMap.put("value", value);
+        List<Map<String, Number>> data = new ArrayList<>(1);
+        data.add(dataMap);
+        Map<String, Object> outer = new HashMap<>(4);
```

Comment:
Yes, you are right. Having default fill factor 0.75, 4 was the smallest initial capacity for me that could hold 2 elements without rehashing. Firstly, 3 would actually suffice too, as {{3 * 0.75 = 2.25}} (which I was not able to eval in my head before) and secondly, HashMap in Java 8 seems to work differently than before, meaning that initial capacity of 2 would suffice on this place too.